Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -433,25 +388,27 @@ extension ArcGISARView: SCNSceneRendererDelegate {
     }
     
     public func renderer(_ renderer: SCNSceneRenderer, willRenderScene scene: SCNScene, atTime time: TimeInterval) {
+        // If we haven't started yet, return.
+        guard notifiedStartOrFailure else { return }
         
         // Get transform from SCNView.pointOfView.
         guard let transform = arSCNView.pointOfView?.transform else { return }
         let cameraTransform = simd_double4x4(transform)
         
-        let finalQuat:simd_quatd = simd_mul(simd_mul(compensationQuat, simd_quatd(cameraTransform)), orientationQuat)
-        var transformationMatrix = AGSTransformationMatrix(quaternionX: finalQuat.vector.x,
+        // Calculate our final quaternion and create the new transformation matrix.
+        let finalQuat:simd_quatd = simd_mul(compensationQuat, simd_quatd(cameraTransform))
+        let transformationMatrix = AGSTransformationMatrix(quaternionX: finalQuat.vector.x,
                                                            quaternionY: finalQuat.vector.y,
                                                            quaternionZ: finalQuat.vector.z,
                                                            quaternionW: finalQuat.vector.w,
                                                            translationX: (cameraTransform.columns.3.x) * translationTransformationFactor,
                                                            translationY: (-cameraTransform.columns.3.z) * translationTransformationFactor,
                                                            translationZ: (cameraTransform.columns.3.y) * translationTransformationFactor)
         
-        transformationMatrix = initialTransformationMatrix.addTransformation(transformationMatrix)
-        
-        let camera = AGSCamera(transformationMatrix: transformationMatrix)
-        sceneView.setViewpointCamera(camera)
+        // Set the matrix on the camera controller.
+        cameraController.transformationMatrix = transformationMatrix
         
+        // Render the Scene with the new transformation.
         sceneView.renderFrame()
```

Comment:
Does the camera controller not do this automatically?

Reply:
No, it is up to the client code to do the rendering.